src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs

```diff
@@ -418,12 +420,28 @@ private IEnumerable<Exception> KillTree(SafeProcessHandle handle)
             return exceptions;
         }
 
-        private IReadOnlyList<(Process Process, SafeProcessHandle Handle)> GetProcessHandlePairs(Func<Process, bool> predicate)
+        private List<(Process Process, SafeProcessHandle Handle)> GetProcessHandlePairs(Func<Process, bool> predicate)
         {
-            return GetProcesses()
-                .Select(p => (Process: p, Handle: SafeGetHandle(p)))
-                .Where(p => !p.Handle.IsInvalid && predicate(p.Process))
-                .ToList();
+            var results = new List<(Process Process, SafeProcessHandle Handle)>();
+
+            foreach (Process p in GetProcesses())
+            {
+                SafeProcessHandle h = SafeGetHandle(p);
+                if (!h.IsInvalid)
+                {
+                    if (predicate(p))
+                    {
+                        results.Add((p, h));
+                    }
+                    else
+                    {
+                        p.Dispose();
```

Comment:
Took me a little while to figure out why these `Dispose` calls were still safe, but LGTM. :)

Reply:
This is an example where LINQ lets you write concise, incorrect code ;)